### PR TITLE
Fixing floating point errors

### LINKED ISSUE #31: 0.1+ per second problem
I am trying to change this block of code:

```
upgrade('Upgrade1', 15, 1)
```
to this:
```
upgrade('Upgrade1', 15, 0.1)
```
The way this code works is that it calls another function with the parameters of ```name```, ```price```, and ```money```. The problem is that if I do this instead of adding 0.1 **P**er **S**econd it adds 0.100000000001 PS(see image).

 
image

### PR DESCRIPTION
**Pull request by giovannipessiva**
Hello, I implemented a function which should fix #31 
Hope it helps! 🍺

### CODE REVIEW

**Comment by StevenCopeland** [comment]
Hello, @giovannipessiva! This is your first Pull Request that will be reviewed by [Ebert](https://ebertapp.io/), an automatic Code Review service. It will leave comments on this diff with potential issues and style violations found in the code as you push new commits. You can also see all the issues found on this Pull Request on its [review page](https://ebertapp.io/github/StevenCopeland/FILEPLAN/pulls/32). Please check our [documentation](https://docs.ebertapp.io) for more information.

File: script.js [hunk 1]
```diff
@@ -241,9 +241,24 @@ var idea = function(name, cost, addition) {
...
+ */
+function sumDecimals(n1, n2) {
+	a = n1+"";
+	b = n2+"";
+	let aIntegerPart = a.indexOf(".") !== -1? a.indexOf(".") - 1 : a.length; 
+	let aFractalPart = a.indexOf(".") !== -1? (a.length - a.indexOf(".")) : 0;
+	let bIntegerPart = b.indexOf(".") !== -1? b.indexOf(".") - 1 : b.length; 
+	let bFractalPart = b.indexOf(".") !== -1? (b.length - b.indexOf(".")) : 0;
+	let maxLength = Math.max(aIntegerPart,bIntegerPart) + Math.max(aFractalPart,bFractalPart);
+	return Number((n1 + n2).toPrecision(maxLength)));
```
Comment (StevenCopeland) [inline script.js]: Parsing error: Unexpected token )

**Comment by StevenCopeland** [comment]
Ebert has finished reviewing this Pull Request and has found:

* 1 possible new issue (including those that may have been commented here).
* 17 fixed issues! 🎉

You can see more details about this review at https://ebertapp.io/github/StevenCopeland/FILEPLAN/pulls/32.

File: script.js [hunk 2]
```diff
@@ -241,9 +241,24 @@ var idea = function(name, cost, addition) {
...
+	let bIntegerPart = b.indexOf(".") !== -1? b.indexOf(".") - 1 : b.length; 
+	let bFractalPart = b.indexOf(".") !== -1? (b.length - b.indexOf(".")) : 0;
+	let maxLength = Math.max(aIntegerPart,bIntegerPart) + Math.max(aFractalPart,bFractalPart);
+	return Number((n1 + n2).toPrecision(maxLength));
+}
+
 var upgrade = function(name, cost, addition) {
 	if (clicker.bytes >= cost) {
-		clicker.perSec += addition;
+		clicker.perSec = sumDecimals(clicker.perSec, addition);
```
Comment (StevenCopeland) [inline script.js]: Expected indentation of 2 spaces but found 2 tabs.

[Read more about it here.](http://eslint.org/docs/rules/indent)

File: script.js [hunk 3]
```diff
@@ -241,9 +241,24 @@ var idea = function(name, cost, addition) {
...
+ * fixing floating point errors
+ */
+function sumDecimals(n1, n2) {
+	a = n1+"";
+	b = n2+"";
+	let aIntegerPart = a.indexOf(".") !== -1? a.indexOf(".") - 1 : a.length; 
+	let aFractalPart = a.indexOf(".") !== -1? (a.length - a.indexOf(".")) : 0;
+	let bIntegerPart = b.indexOf(".") !== -1? b.indexOf(".") - 1 : b.length; 
+	let bFractalPart = b.indexOf(".") !== -1? (b.length - b.indexOf(".")) : 0;
+	let maxLength = Math.max(aIntegerPart,bIntegerPart) + Math.max(aFractalPart,bFractalPart);
```
Comment (StevenCopeland) [inline script.js]: Expected indentation of 2 spaces but found 1 tab.

[Read more about it here.](http://eslint.org/docs/rules/indent)

File: script.js [hunk 4]
```diff
@@ -241,9 +241,24 @@ var idea = function(name, cost, addition) {
...
+ * Return the sum of two numbers,
+ * fixing floating point errors
+ */
+function sumDecimals(n1, n2) {
+	a = n1+"";
+	b = n2+"";
+	let aIntegerPart = a.indexOf(".") !== -1? a.indexOf(".") - 1 : a.length; 
+	let aFractalPart = a.indexOf(".") !== -1? (a.length - a.indexOf(".")) : 0;
+	let bIntegerPart = b.indexOf(".") !== -1? b.indexOf(".") - 1 : b.length; 
+	let bFractalPart = b.indexOf(".") !== -1? (b.length - b.indexOf(".")) : 0;
```
Comment (StevenCopeland) [inline script.js]: Strings must use singlequote.

[Read more about it here.](http://eslint.org/docs/rules/quotes)

Comment (StevenCopeland) [inline script.js]: Possible undefined variable: 'b' is not defined.

If this is intended to be a global variable, you can supress this warning by adding a  `/*global [VARIABLE NAME] */` comment on this file or set it under the `globals` section of your `.eslintrc` configuration.

[Read more about it here.](http://eslint.org/docs/rules/no-undef)

Comment (StevenCopeland) [inline script.js]: Possible undefined variable: 'b' is not defined.

If this is intended to be a global variable, you can supress this warning by adding a  `/*global [VARIABLE NAME] */` comment on this file or set it under the `globals` section of your `.eslintrc` configuration.

[Read more about it here.](http://eslint.org/docs/rules/no-undef)

Comment (StevenCopeland) [inline script.js]: Gratuitous parentheses around expression.

Comment (StevenCopeland) [inline script.js]: Strings must use singlequote.

[Read more about it here.](http://eslint.org/docs/rules/quotes)

Comment (StevenCopeland) [inline script.js]: Possible undefined variable: 'b' is not defined.

If this is intended to be a global variable, you can supress this warning by adding a  `/*global [VARIABLE NAME] */` comment on this file or set it under the `globals` section of your `.eslintrc` configuration.

[Read more about it here.](http://eslint.org/docs/rules/no-undef)

Comment (StevenCopeland) [inline script.js]: Expected indentation of 2 spaces but found 1 tab.

[Read more about it here.](http://eslint.org/docs/rules/indent)

File: script.js [hunk 5]
```diff
@@ -241,9 +241,24 @@ var idea = function(name, cost, addition) {
...
+/**
+ * Return the sum of two numbers,
+ * fixing floating point errors
+ */
+function sumDecimals(n1, n2) {
+	a = n1+"";
+	b = n2+"";
+	let aIntegerPart = a.indexOf(".") !== -1? a.indexOf(".") - 1 : a.length; 
+	let aFractalPart = a.indexOf(".") !== -1? (a.length - a.indexOf(".")) : 0;
+	let bIntegerPart = b.indexOf(".") !== -1? b.indexOf(".") - 1 : b.length;
```
Comment (StevenCopeland) [inline script.js]: Possible undefined variable: 'b' is not defined.

If this is intended to be a global variable, you can supress this warning by adding a  `/*global [VARIABLE NAME] */` comment on this file or set it under the `globals` section of your `.eslintrc` configuration.

[Read more about it here.](http://eslint.org/docs/rules/no-undef)

Comment (StevenCopeland) [inline script.js]: Strings must use singlequote.

[Read more about it here.](http://eslint.org/docs/rules/quotes)

Comment (StevenCopeland) [inline script.js]: Possible undefined variable: 'b' is not defined.

If this is intended to be a global variable, you can supress this warning by adding a  `/*global [VARIABLE NAME] */` comment on this file or set it under the `globals` section of your `.eslintrc` configuration.

[Read more about it here.](http://eslint.org/docs/rules/no-undef)

Comment (StevenCopeland) [inline script.js]: Strings must use singlequote.

[Read more about it here.](http://eslint.org/docs/rules/quotes)

Comment (StevenCopeland) [inline script.js]: Possible undefined variable: 'b' is not defined.

If this is intended to be a global variable, you can supress this warning by adding a  `/*global [VARIABLE NAME] */` comment on this file or set it under the `globals` section of your `.eslintrc` configuration.

[Read more about it here.](http://eslint.org/docs/rules/no-undef)

Comment (StevenCopeland) [inline script.js]: Expected indentation of 2 spaces but found 1 tab.

[Read more about it here.](http://eslint.org/docs/rules/indent)

File: script.js [hunk 6]
```diff
@@ -241,9 +241,24 @@ var idea = function(name, cost, addition) {
...
 
+/**
+ * Return the sum of two numbers,
+ * fixing floating point errors
+ */
+function sumDecimals(n1, n2) {
+	a = n1+"";
+	b = n2+"";
+	let aIntegerPart = a.indexOf(".") !== -1? a.indexOf(".") - 1 : a.length; 
+	let aFractalPart = a.indexOf(".") !== -1? (a.length - a.indexOf(".")) : 0;
```
Comment (StevenCopeland) [inline script.js]: Strings must use singlequote.

[Read more about it here.](http://eslint.org/docs/rules/quotes)

Comment (StevenCopeland) [inline script.js]: Possible undefined variable: 'a' is not defined.

If this is intended to be a global variable, you can supress this warning by adding a  `/*global [VARIABLE NAME] */` comment on this file or set it under the `globals` section of your `.eslintrc` configuration.

[Read more about it here.](http://eslint.org/docs/rules/no-undef)

Comment (StevenCopeland) [inline script.js]: Possible undefined variable: 'a' is not defined.

If this is intended to be a global variable, you can supress this warning by adding a  `/*global [VARIABLE NAME] */` comment on this file or set it under the `globals` section of your `.eslintrc` configuration.

[Read more about it here.](http://eslint.org/docs/rules/no-undef)

Comment (StevenCopeland) [inline script.js]: Gratuitous parentheses around expression.

Comment (StevenCopeland) [inline script.js]: Strings must use singlequote.

[Read more about it here.](http://eslint.org/docs/rules/quotes)

**Comment by StevenCopeland** [comment]
Ebert has finished reviewing this Pull Request and has found:

* 22 possible new issues (including those that may have been commented here).

You can see more details about this review at https://ebertapp.io/github/StevenCopeland/FILEPLAN/pulls/32.

**Comment by StevenCopeland** [comment]
@giovannipessiva Sorry about the ebert and Codacy reviews!

**Comment by StevenCopeland** [comment]
Ebert has finished reviewing this Pull Request and has found:

* 2 possible new issues (including those that may have been commented here).

You can see more details about this review at https://ebertapp.io/github/StevenCopeland/FILEPLAN/pulls/32.

**Comment by StevenCopeland** [comment]
@giovannipessiva They both contradict each other! Ebert wants single quotes and Codacy wants double quotes.

**Comment by giovannipessiva** [comment]
@StevenCopeland well, everyone has their own tastes :-)
I am submitting an additional PR to fix an edge case

**Pull request merged**